app/Foliage/PrepareSource.hs

```diff
@@ -120,3 +88,21 @@ addPrepareSourceRule inputDir cacheDir = addBuiltinRule noLint noIdentity run
             liftIO $ rewritePackageVersion cabalFilePath pkgVersion
 
           return $ RunResult ChangedRecomputeDiff BS.empty srcDir
+
+    extractFromTarball tarballPath mSubdir outDir = do
+      withTempDir $ \tmpDir -> do
+        cmd_ "tar xzf" [tarballPath] "-C" [tmpDir]
```

Comment:
Why not having a `tarInvocation` thing (like you do with `curlInvocation`) with code comment that recall what each CLI flag does and what we need them?